liferay-echo/liferay-portal
# LPS-179146 Add ability to send email notifications when web content expires

## Motivation

[Link to ticket](https://issues.liferay.com/browse/LPS-179146)

Based on a feature request, one client wants to receive notifications for expired web contents. This PR adds support for this

## Change summary:

* Adds e-mail templates for expired web contents, and views in configuration
* Adds support for notification for expired web contents


## Test Scenario 1

* Enable FF like this: feature.flag.LPS-179142=true
* Configure SMTP server to receive e-mails (you can use mailHog), also un-blacklist test@liferay.com from mail.send.blacklist
* Go to Content & Data > Web Content and create a new folder 
* Click on the Folder to navigate to it
* Open the side bar and subscribe to it
* Create a new Web Content and publish it
* Create a new user and impersonate it
* Go to the folder that was created with the first user
* Click on the kebab and expire it
* Go back to the other user, and a notification should be shown, and an e-mail will also be sent

## Test Scenario 1

* Enable FF like this: feature.flag.LPS-179142=true
* Configure SMTP server to receive e-mails (you can use mailHog), also un-blacklist test@liferay.com from mail.send.blacklist
* Adjust Control Panel>System Settings>Web Content and change the check interval from 15 minutes to 1
* Go to Content & Data > Web Content and create a new folder 
* Click on the Folder to navigate to it
* Open the side bar and subscribe to it
* Create a new Web Content
* Set the expiration date to 2 minutes in the future and publish
* Wait for a while, and when you refresh a notification should be shown, and an e-mail will also be sent
